### PR TITLE
ADD: Added verification functionality for checking if user is owner

### PR DESCRIPTION
**Pull request by Foopert**

### CODE REVIEW

File: db/db.js [hunk 1]
```diff
@@ -55,5 +55,26 @@ const createSection = async (attrs) => {
     throw Error(`Section is referencing Project ${attrs.project} which doesn't exist.`)
 }
 
+const createTask = async (attrs) => {
+    // Todo: validate you are an owner of the project
+    const sectionRef = findRecord(Section, attrs.section)
+    if (sectionRef) {
+        return await saveRecord(Section, attrs.section)
```
Comment (NickLewanowicz) [inline db/db.js]: 'createTask' should be saving a 'Task' model not a 'Section' model. Also you need to pass ALL the attrs not just section id. Should look like this:

return await saveRecord(Task, attrs)

File: graphql/resolvers.js [hunk 2]
```diff
@@ -124,10 +124,14 @@ const resolvers = (models) => ({
...
     },
 
-    createTask(root, args) {
-      const task = new models.Task(args);
-      task.created = Math.floor(new Date() / 1000)
-      return task.save().then((response) => response);
+    async createTask(root, args, context) {
+      const userId = getUserId(context)
+      if (await db.userOwnsSection(args.section, userId)) {
+        const task = new models.Task(args);
```
Comment (NickLewanowicz) [inline graphql/resolvers.js]: what is inside of this conditional should be moved to the db.createTask, would look like this:
if (await db.userOwnsSection(args.section, userId)) {
    return await db.createTask(attrs)
}

then in createTask you would set the `created` attribute and return the savedRecord

File: utils.js [hunk 3]
```diff
@@ -8,7 +8,7 @@ function getUserId(context) {
     const { userId } = jwt.verify(token, APP_SECRET)
     return userId
   }
-  return ""
+  return userId
```
Comment (NickLewanowicz) [inline utils.js]: REMOVE THIS. If the user isnt authorized this should return empty

File: graphql/resolvers.js (hunk 2 above)
Comment (Foopert) [reply graphql/resolvers.js]: attrs isnt defined in resolvers.js, should it be args instead?

**Comment by Foopert** [comment]
- Fixed functionality for createTask function, now passes all args and saves record with attrs
- Removed userId in utils.js (big vulnerability, was only used for testing but I missed it, thanks for the catch)

**Pull request merged**